src/main/java/com/iexec/sms/tee/EnableIfTeeProvider.java

```diff
@@ -0,0 +1,70 @@
+package com.iexec.sms.tee;
+
+import com.iexec.common.tee.TeeEnclaveProvider;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+import org.springframework.core.type.AnnotationMetadata;
+
+/**
+ * Define a way to include beans only if a profile is enabled
+ * <p/>
+ * Usage:
+ * Annotate any bean with the following:
+ * {@code
+ * @Conditional(EnableIfTeeProvider.class)
+ * @EnableIfTeeProviderDefinition(providers = TeeEnclaveProvider.<member>)
+ * }
+ * <p/>
+ * If bean is not annotated with {@link EnableIfTeeProviderDefinition}
+ * or {@link EnableIfTeeProviderDefinition#providers()} is null or empty,
+ * the bean won't be loaded.
+ */
+@Slf4j
+public class EnableIfTeeProvider implements Condition {
+    @Override
+    public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
+        final String[] activeProfiles = context.getEnvironment().getActiveProfiles();
+        final String beanClassName = ((AnnotationMetadata) metadata).getClassName();
+        final Class<?> clazz;
+        try {
+            clazz = Class.forName(beanClassName);
+        } catch (ClassNotFoundException e) {
+            log.error(
+                    "Bean does not exist [bean: {}]",
```

Comment:
very minor
```suggestion
                    "Bean does not exist [bean:{}]",
```
(everywhere :))

Reply:
See https://github.com/iExecBlockchainComputing/iexec-sms/pull/119/commits/41a06cd3457e0e7f0f86924a5ac0429d6c60baeb.